provisioning/utils.go

```diff
@@ -47,11 +48,21 @@ func getPodHostIP(podClient coreclientv1.PodsGetter, targetNamespace string) (st
 	return hostIP, err
 }
 
-func GetIronicIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec) (string, error) {
-	// if UseIronicProxy(config) {
-	// TODO: API VIP ???
-	// } else
-	if config.ProvisioningNetwork != metal3iov1alpha1.ProvisioningNetworkDisabled && !config.VirtualMediaViaExternalNetwork {
+func GetIronicIP(client kubernetes.Interface, targetNamespace string, config *metal3iov1alpha1.ProvisioningSpec, osclient osclientset.Interface) (string, error) {
+	if UseIronicProxy(config) {
+		ctx := context.Background()
+		infra, err := osclient.ConfigV1().Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+		if err != nil {
+			err = fmt.Errorf("Unable to get /apis/config.openshift.io/v1/infrastructures/cluster: %w", err)
+			return "", err
+		}
+		if infra.Status.PlatformStatus != nil {
+			return infra.Status.PlatformStatus.BareMetal.APIServerInternalIP, nil
+		} else {
+			err = fmt.Errorf("Unable to get /apis/config.openshift.io/v1/infrastructures/cluster/status/platformStatus")
```

Comment:
The URL probably makes little sense for readers, something like `Cannot detect bare metal API VIP: not a baremetal platform`